### PR TITLE
Use ParseDevfileAndValidate instead of ParseAndValidate

### PR DESCRIPTION
**Pull request by dharmit**
**What type of PR is this?**
/kind cleanup

**What does this PR do / why we need it**:
ParseAndValidate has been deprecated in favour of
ParseDevfileAndValidate

**Which issue(s) this PR fixes**:

Fixes NA

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
Test should pass

### CODE REVIEW

**Comment by dharmit** [comment]
Not sufficient in current form. odo code has calls to `ValidateDevfileData` which need to be removed since `ParseDevfileAndValidate` does both parsing and validation. 

/hold work-in-progress

File: pkg/odo/cli/catalog/describe/component.go [hunk 1]
```diff
@@ -212,16 +213,12 @@ func GetDevfile(devfileComponent catalog.DevfileComponentType) (parser.DevfileOb
...
+		devObj, err = devfile.ParseDevfileAndValidate(parser.ParserArgs{URL: registryURL.String()})
 		if err != nil {
 			return devObj, errors.Wrapf(err, "Failed to download devfile.yaml from OCI-based registry for devfile component: %s", devfileComponent.Name)
 		}
 	}
 
-	err = validate.ValidateDevfileData(devObj.Data)
-	if err != nil {
-		return devObj, err
-	}
```
Comment (mik-dass) [inline pkg/odo/cli/catalog/describe/component.go:224]: This call is for performing `odo` specific validation and I think it shouldn't be removed.

Comment (dharmit) [reply pkg/odo/cli/catalog/describe/component.go:224]: There are 9 occurrences of its removal in my PR. It looks like all of them are odo specific validations. I feel I should undo all of them. WDYT?

Comment (mik-dass) [reply pkg/odo/cli/catalog/describe/component.go:224]: Yes, we need to undo all of them.

Comment (dharmit) [reply pkg/odo/cli/catalog/describe/component.go]: OK. So this wasn't necessary - https://github.com/openshift/odo/pull/4629#issuecomment-820935598 😂

Comment (dharmit) [reply pkg/odo/cli/catalog/describe/component.go]: I have removed the second commit so all those changes have been removed.

**Comment by dharmit** [comment]
Looks like infra error to me:
```
ERRO[2021-04-16T08:31:21Z] level=error                                  
ERRO[2021-04-16T08:31:21Z] level=error msg=Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api.ci-op-8zqhbqy6-ba78b.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists] 
ERRO[2021-04-16T08:31:21Z] level=error msg=	status code: 400, request id: e1a299a0-cbb2-4159-92d5-b5d75d1e1015 
ERRO[2021-04-16T08:31:21Z] level=error                                  
ERRO[2021-04-16T08:31:21Z] level=error msg=  on ../tmp/openshift-install-314555287/route53/base.tf line 36, in resource "aws_route53_record" "api_external_alias": 
ERRO[2021-04-16T08:31:21Z] level=error msg=  36: resource "aws_route53_record" "api_external_alias" { 
ERRO[2021-04-16T08:31:21Z] level=error                                  
ERRO[2021-04-16T08:31:21Z] level=error                                  
ERRO[2021-04-16T08:31:21Z] level=fatal msg=failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to apply Terraform: failed to complete the change 
```
/test v4.7-integration-e2e

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**